### DISCUSSION TITLE
Promise: bluebird error

### DISCUSSION THREAD

**Discussion by Loveyless** [sidorares/node-mysql2#1579] [General]
(MySQL2 使用范围内可用的默认 Promise 对象。 但是你可以选择你想使用的 Promise 实现) in documentation_zh-cn ,maybe it doesn't work
```
// get the client
const mysql = require('mysql2/promise');

// get the promise implementation, we will use bluebird
const bluebird = require('bluebird');

// create the connection, specify bluebird as Promise
const connection = await mysql.createConnection({host:'localhost', user: 'root',password:'root', database: 'test', Promise: bluebird });

// query database
const [rows, fields] = await connection.execute('SELECT * FROM products p WHERE p.id = ?', [1]);
```

SyntaxError: await is only valid in async functions and the top level bodies of 
modules

I have npm i bluebird

Thank you very much

**Comment by sidorares** [comment]
you need to run your script as ES module in order to be able to use top level await. ( change extension to `.mjs` or mark it as module in package.json )

https://www.stefanjudis.com/today-i-learned/top-level-await-is-available-in-node-js-modules/

**Reply by Loveyless** [reply]
Thank you very much!
